core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -124,20 +133,19 @@ public static ListNamespacesResponse listNamespaces(
       SupportsNamespaces catalog, Namespace parent, String pageToken, String pageSize) {
     List<Namespace> results;
     List<Namespace> subResults;
+    String nextToken = null;
 
     if (parent.isEmpty()) {
       results = catalog.listNamespaces();
     } else {
       results = catalog.listNamespaces(parent);
     }
 
-    int start = INTIAL_PAGE_TOKEN.equals(pageToken) ? 0 : Integer.parseInt(pageToken);
-    int end = start + Integer.parseInt(pageSize);
-    subResults = results.subList(start, end);
-    String nextToken = String.valueOf(end);
+    int pageStart = INITIAL_PAGE_TOKEN.equals(pageToken) ? 0 : Integer.parseInt(pageToken);
+    subResults = paginate(results, pageStart, Integer.parseInt(pageSize));
 
-    if (end >= results.size()) {
-      nextToken = null;
+    if (pageStart + subResults.size() < results.size()) {
+      nextToken = String.valueOf(pageStart + subResults.size());
     }
```

Comment:
Is it possible to put all this in a generic method `private static <T> String nextToken(List<T> results)` that way we don't have to duplicate and can more easily reuse it for other cases in the future?

Reply:
@amogh-jahagirdar pushed a new version just now. This uses a `Pair` to make pagination more coincise. Not a big fan of the bloat in generics that comes with it, but that it just my person taste. It does make the calls more concise for namespaces/tables/views.

The only thing I thought about was maybe use FQN for the `Pair` object I am using since the is a Iceberg internal class with the same name. The latter looked like a different animal than a simple Tuple.

Let me know your thoughts on this approach.

Reply:
Thanks @rcjverhoef! Yeah ideally we could avoid the generics complexity but I do think it's worth it. My main rationale for the separate helper method was that future APIs which need pagination don't have the risk of duplicating the pagination logic, possibly incorrectly etc 

>The only thing I thought about was maybe use FQN for the Pair object I am using since the is a Iceberg internal class with the same name. The latter looked like a different animal than a simple Tuple.

The logic of the helper method is all good to me, but I do think we should be using the Iceberg pair and not the apache commons pair implementation. It's true that the Iceberg pair implements a few interfaces beyond just being a normal pair, but throughout the code there are many places where it is just used as a plain old pair, and we don't use the apache commons one. So I think it makes more sense to just optimize for consistency here.

Reply:
Consistency is a good argument. Code adjusted!